shardingsphere-scaling/shardingsphere-scaling-dialect/shardingsphere-scaling-opengauss/src/main/java/org/apache/shardingsphere/scaling/opengauss/wal/decode/MppdbDecodingPlugin.java

```diff
@@ -0,0 +1,247 @@
+        if (columnType.startsWith("numeric")) {
+            return new BigDecimal(data);
+        }
+        if (columnType.startsWith("bit") || columnType.startsWith("bit varying")) {
+            return data;
+        }
+        switch (columnType) {
+            case "smallint":
+                return Short.parseShort(data);
+            case "integer":
+                return Integer.parseInt(data);
+            case "bigint":
+                return Long.parseLong(data);
+            case "real":
+                return Float.parseFloat(data);
+            case "double precision":
+                return Double.parseDouble(data);
+            case "boolean":
+                return Boolean.parseBoolean(data);
+            case "time without time zone":
+                try {
+                    return timestampUtils.toTime(null, data);
+                } catch (final SQLException ex) {
+                    throw new DecodingException(ex);
+                }
+            case "date":
+                return Date.valueOf(data);
+            case "timestamp without time zone":
+                try {
+                    return timestampUtils.toTimestamp(null, data);
+                } catch (final SQLException ex) {
+                    throw new DecodingException(ex);
+                }
+            case "bytea":
+                return decodeHex(data.substring(2));
+            case "character varying":
+                return decodeString(data);
+            default:
+                return data;
+        }
+    }
+    
+    private static String decodeString(final String data) {
+        if (data.length() > 1) {
+            int begin = data.charAt(0) == '\'' ? 1 : 0;
+            int end = data.length() + (data.charAt(data.length() - 1) == '\'' ? -1 : 0);
+            return data.substring(begin, end);
+        }
+        return data;
+    }
+    
+    private byte[] decodeHex(final String hexString) {
+        int dataLength = hexString.length();
+        if (0 != (dataLength & 1)) {
+            throw new IllegalArgumentException(String.format("Illegal hex data %s", hexString));
+        }
+        if (0 == dataLength) {
+            return new byte[0];
+        }
+        byte[] result = new byte[dataLength >>> 1];
+        for (int i = 0; i < dataLength; i += 2) {
+            result[i >>> 1] = decodeHexByte(hexString, i);
+        }
+        return result;
+    }
+    
+    private byte decodeHexByte(final String hexString, final int index) {
+        int firstHexChar = Character.digit(hexString.charAt(index), 16);
+        int secondHexChar = Character.digit(hexString.charAt(index + 1), 16);
+        if (-1 == firstHexChar || -1 == secondHexChar) {
+            throw new IllegalArgumentException(String.format("Illegal hex byte '%s' in index %d", hexString, index));
+        }
+        return (byte) ((firstHexChar << 4) + secondHexChar);
+    }
+    
+    @Setter
+    @Getter
+    public static final class MppTableData {
```

Comment:
It's used much times outside of this plugin class, could we put it as dividual java file?